### PR TITLE
Fix viewer user handling by Carto::Api::VisualizationsController

### LINKED ISSUE #12379: No check for viewer in Carto::Visualization
As a viewer, using the API, you can create/update/delete visualizations. We used to raise errors on those cases (Member model), but we have lost it during the refactor (which also suggests a lack of tests).

cc @juanignaciosl

### PR DESCRIPTION
**Pull request by juanignaciosl**
This fixes #12379

### CODE REVIEW

File: spec/models/carto/visualization_spec.rb [hunk 1]
```diff
@@ -298,6 +298,16 @@ def build_visualization(attrs = {})
     end
   end
 
+  describe 'creation' do
+    it 'is not valid if user is viewer' do
+      viewer = FactoryGirl.build(:carto_user, viewer: true)
+      visualization = FactoryGirl.build(:carto_visualization, user: viewer)
+      visualization.valid?.should be_false
```
Comment (mayoral) [inline spec/models/carto/visualization_spec.rb]: Can we use `expect` syntax for these tests? Mainly thinking about a future migration to RSpec 3.

File: spec/requests/carto/api/visualizations_controller_spec.rb [hunk 2]
```diff
@@ -1895,6 +1895,26 @@ def first_data_layer_from_response(response)
...
+              visChanges: 0,
+              name: "untitled_table_XXX_map"
+            }
+
+            @org_user_1.viewer = true
+            @org_user_1.save
+
+            post_json(api_v1_visualizations_create_url(user_domain: @org_user_1.username, api_key: @org_user_1.api_key),
+                      payload) do |response|
+              response.status.should eq 403
```
Comment (mayoral) [inline spec/requests/carto/api/visualizations_controller_spec.rb]: Same here about `expect`.

**Comment by juanignaciosl** [comment]
@mayoral as we haven't agreed upon `expect` syntax, I think that we shouldn't move an issue backwards in the kanban because of it. Feel free to use it from now on, and I'll do it too (when I remember :_) ), but let's move this forward if there's nothing else that you'd change.

**Comment by mayoral** [comment]
👍

**Pull request merged**